cmd/idgenerators.go

```diff
@@ -101,3 +101,35 @@ func (m NeLoIdGenerator) GenerateId(c *gin.Context) {
 	})
 	log.Printf("Successfully generated the NeLo '%s'", neloId)
 }
+
+// MeLoIdGenerator is an IdGenerator that generates MeLo-IDs (Messlokation-IDs)
+type MeLoIdGenerator struct{}
+
+var numbers = []rune("0123456789")
+var allowedMeLoCharacters = []rune("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+// GenerateId of the MeLoIdGenerator returns a new random, 33 character melo-id; MeLo-IDs have no checksum
+func (m MeLoIdGenerator) GenerateId(c *gin.Context) {
+	// See VDE-AR-N 4400 https://www.vde-verlag.de/normen/0400343/vde-ar-n-4400-anwendungsregel-2019-07.html
+	/*              DE|001069|66646|10000000000000012345
+	                 |     |      |        |
+	  Landesziffern -|     |      |        |- Laufende Nummer
+	                       |      |
+	Netzbetreibernummer ---|      |-- PLZ
+	*/
```

Comment:
sehr schön ;)

Reply:
nur ausm wiki kopiert. all credits to @hf-krechan